src/realm/mixed.hpp

```diff
@@ -694,11 +694,8 @@ inline BinaryData Mixed::get<BinaryData>() const noexcept
 {
     if (is_null())
```

Comment:
Good eye. For some reason we do this for strings and binaries only. We could change it but then it is the responsibility of the caller to always check for nulls. I guess we do this because both StringData and BinaryData store null internally as opposed to our other types that are used with `std::optional` and this design conflicts with the Mixed type which is supposed to store Null or a StringData which itself can be null. Perhaps we can solve this in a different PR.